### ISSUE TITLE
direnv.toml file is ignored

### ISSUE THREAD

**Issue by clebs** [direnv/direnv#1344]
**Describe the bug**
After creating the direnv config file and adding settings under `~/.config/direnv/direnv.toml` they are not being applied. 

**To Reproduce**
- Install direnv via `nix-darwin`: `programs.direnv.enable = true;`
- Create a config file under `~/.config/direnv/direnv.toml`
- Add a setting like:
```
[global]
load_dotenv = true
disable_stdin = true
```

**Expected behavior**
When running `direnv status` disable stdin should be set to true and `.env` files should be loaded.

**Environment**
 - OS: macOS 15.1 Sequoia
 - Shell: zsh
 - Direnv version: 2.35.0

**Comment by jdmallen** [comment]
Also finding this to be true using 2.32.1 (with Ubuntu 24.04.1). I placed the file exactly where [the instructions say](https://direnv.net/man/direnv.toml.1.html), but it's still showing diffs and pushing to `stdin`. The whitelist seems to work, though. 

```sh
 % cat $XDG_CONFIG_HOME/direnv/direnv.toml
[global]
disable_stdin = true
load_dotenv = true
hide_env_diff = true

[whitelist]
prefix = ["/home/jesse/git"]
```

Looking at `config.go` [here](https://github.com/direnv/direnv/blob/23c1f925d843ef4575637602ea5278a09cb0d93c/internal/cmd/config.go#L45) it seems to suggest that getting rid of the `[global]` header will work for those keys, but removing it does not fix the functionality.

I can confirm, though, that `direnv` _is seeing_ the config file, as when I change the values for the global keys from `true` to `"true"`, I get this (newlines added), presumably being [thrown from here](https://github.com/BurntSushi/toml/blob/092fca1af5f28b5de952df96390b41db66150ffc/decode.go#L538):

```sh
direnv: error LoadConfig() failed to parse /home/jesse/.config/direnv/direnv.toml:
toml: line 1 (last key "disable_stdin"):
incompatible types: TOML value has type string; destination has type boolean
```

Hopefully this helps someone who is better than Go than I am track down why the `[global]` and table-less keys are being ignored.

**Comment by jdmallen** [comment]
UPDATE: I tried installing the latest version (v2.35.0) and it appears to (mostly) respect the config file now.

[The fix is somewhere in this diff 😅](https://github.com/direnv/direnv/compare/v2.32.1..v2.35.0)

It's still writing to stdin, but it's not printing the diff anymore, and it seems to be looking for .env files automatically.

```sh
 % cd ~
direnv: loading ~/.envrc
```

So, not a total win, but a giant step in the right direction.

**Comment by plaukiu** [comment]
same here, 2.35.0, config changes are haphazard. I managed to get `hide_env_diff = true` to work somehow, but now changing anything does nothing at all, malformed config gives out no errors.

**Comment by reshen** [comment]
tl;dr -if you're using Nix, this post may help.

> Install direnv via nix-darwin

Not sure if it applies to nix-darwin, but TIL that `~/.config/direnv/direnv.toml` is completely ignored on NixOS as I enabled direnv via [nix's direnv option](https://search.nixos.org/options?channel=25.05&show=programs.direnv.enable&from=0&size=50&sort=relevance&type=packages&query=direnv). Turns out configuring direnv via nix options does work. Here's a complete example from my `configuration.nix`:

```nix
  # Before each prompt, direnv checks for the existence of a .envrc file (and
  # optionally a .env file) in the current and parent directories. If the file
  # exists (and is authorized), it is loaded into a bash sub-shell and all
  # exported variables are then captured by direnv and then made available to
  # the current shell.
  programs.direnv.enable = true;
  programs.direnv.enableFishIntegration = true;
  programs.direnv.settings = {
    global = {
      # Hide the diff of the environment variables when
      # loading the .envrc. Defaults to false.
      hide_env_diff = true;

      # Sets the log format for direnv outputs. Set to “-” to disable normal logging.
      log_format = "-";
      log_filter = "^$";
    };
    whitelist = {
      exact = [
        # Accepts an array of strings. Each string can be a directory name or the full path to an .envrc file. If a directory name
        # is passed, it will be treated as if it had been passed as itself with /.envrc appended. After resolving the filename,
        # each string will be checked for being an exact match with an .envrc file’s absolute path. If they match exactly, that
        # .envrc file will be implicitly allowed, regardless of contents or past usage of direnv allow or direnv deny.
        "/home/${user}/gamedev/godot-bevy"
        "/home/${user}/gamedev/godot-bevy-main"
      ];
    };
  };
``` 

in practice:
```bash
❯ cd godot-bevy

❯ direnv status | rg Loaded | head -n 2
8:Loaded RC path /home/veris/gamedev/godot-bevy/.envrc
9:Loaded watch: ".envrc" - 2025-07-08T08:32:46-07:00

❯ ls -al ~/.local/share/direnv/allow/
total 8
drwxr-xr-x 2 veris users 4096 Jul  8 08:11 .
drwxr-xr-x 3 veris users 4096 Jul  1 07:11 ..
```

and versions:
```bash
❯ direnv --version
2.36.0
```

**Comment by clebs** [comment]
Thanks for the tip! I will try it out.

Truth be told I switched to packaging `autoenv` on nix myself and now maintain that module and package which does what I want out of the box.

**Comment by juangiordana** [comment]
The [NixOs module](https://github.com/NixOS/nixpkgs/blob/e73de5be04e0eff4190a1432b946d469c794e7b4/nixos/modules/programs/direnv.nix#L149) sets the environment variable `DIRENV_CONFIG` to `/etc/direnv`, so one can manually overwrite that via `~/.zshrc`, `~/.bashrc`, etc.

```sh
# ~/.zshrc
# Read `direnv` settings from ~/.config/direnv/direnv.toml
DIRENV_CONFIG="$XDG_CONFIG_HOME/direnv"
```
